src/lava/lib/optimization/solvers/generic/sub_process_models.py

```diff
@@ -9,22 +9,218 @@
+            growth_params = proc.hyperparameters.get(
+                "growth_schedule_parameters", (3, 2)
+            )
+            beta_growth_indices = proc.hyperparameters.get(
+                "beta_growth_indices", [0]
+            )
+            lr_change = proc.hyperparameters.get("lr_change_type", "indices")
+
+            if backend in CPUS:
+                self.conn_A = Sparse(
+                    weights=csr_matrix(A_pre),
+                    num_message_bits=64,
+                )
+
+                self.conn_A_T = Sparse(
+                    weights=csr_matrix(A_pre.T), num_message_bits=64
+                )
+
+                # Neurons for Constraint Checking
+                self.ProInt = ProportionalIntegralNeuronsPIPGeq(
+                    shape=init_constraints.shape,
+                    constraint_neurons_init=init_constraints,
+                    thresholds=k_pre,
+                    beta=beta,
+                    lr_growth_type=lr_change,
+                    beta_growth_params=growth_params,
+                    beta_growth_indices=beta_growth_indices,
+                )
+            elif backend in NEUROCORES:
+                _, Q_pre_fp_exp = convert_to_fp(Q_pre, 8)
+                A_pre_fp_man, A_pre_fp_exp = convert_to_fp(A_pre, 8)
+                k_pre_fp_man, k_pre_fp_exp = convert_to_fp(k_pre, 24)
+                correction_exp = min(A_pre_fp_exp, Q_pre_fp_exp)
+                A_exp_new = -correction_exp + A_pre_fp_exp
+                A_pre_fp_man = (A_pre_fp_man // 2) * 2
+
+                self.conn_A = Sparse(
+                    weights=csr_matrix(A_pre_fp_man),
+                    num_message_bits=24,
+                )
+
+                self.conn_A_T = Sparse(
+                    weights=csr_matrix(A_pre_fp_man.T),
+                    weight_exp=A_exp_new,
+                    num_message_bits=24,
+                )
+
+                # Neurons for Constraint Checking
+                self.ProInt = ProportionalIntegralNeuronsPIPGeq(
+                    shape=init_constraints.shape,
+                    constraint_neurons_init=init_constraints,
+                    da_exp=A_pre_fp_exp,
+                    thresholds=k_pre_fp_man,
+                    thresholds_exp=k_pre_fp_exp,
+                    beta=beta_man,
+                    beta_exp=beta_exp,
+                    beta_growth_params=growth_params,
+                )
+            else:
+                raise NotImplementedError(str(backend) + BACKEND_MSG)
+
+            proc.in_ports.a_in.connect(self.conn_A.s_in)
+            self.conn_A.a_out.connect(self.ProInt.a_in)
+            self.ProInt.s_out.connect(self.conn_A_T.s_in)
+            self.conn_A_T.a_out.connect(proc.out_ports.s_out)
+            proc.vars.constraint_assignment.alias(
+                self.ProInt.constraint_neuron_state
+            )
+        else:
+            AssertionError("Unknown neuron model specified")
 
 

```

Comment:
Some changes below seem unrelated to QP integration or linting, is this intended?

Reply:
They were moved below because of addition of code above. I have not touched the logic of QUBO